src/prefect/server/models/artifacts.py

```diff
@@ -172,6 +184,44 @@ async def delete_artifact(
     Returns:
         bool: True if the delete was successful, False otherwise
     """
+    current_artifact = await session.get(db.Artifact, artifact_id)
```

Comment:
This might not be the latest artifact though? "current" seems like a weird prefix here. Should this just be `artifact`?